wradlib/georef/raster.py

```diff
@@ -474,3 +482,98 @@ def reproject_raster_dataset(src_ds, **kwargs):
     gdal.ReprojectImage(src_ds, dst_ds, src_srs, dst_srs, resample)
 
     return dst_ds
+
+
+def create_raster_dataset(data, coords, projection=None, nodata=-9999):
+    """ Create In-Memory Raster Dataset
+
+    Parameters
+    ----------
+    data : :class:`numpy:numpy.ndarray`
+        Array of shape (rows, cols) or (bands, rows, cols) containing
+        the data values.
+    coords : :class:`numpy:numpy.ndarray`
+        Array of shape (rows, cols, 2) containing xy-coordinates.
+    projection : osr object
+        Spatial reference system of the used coordinates, defaults to None.
+    nodata : int
+        Value of NODATA
+
+    Returns
+    -------
+    dataset : gdal.Dataset
+        In-Memory raster dataset
+
+    Note
+    ----
+    The origin of the provided data and coordinates is UPPER LEFT.
+    """
+
+    # align data
+    data = data.copy()
+    if data.ndim == 2:
+        data = data[np.newaxis, ...]
+    bands, rows, cols = data.shape
+
+    # create In-Memory Raster with correct dtype
+    mem_drv = gdal.GetDriverByName('MEM')
+    gdal_type = gdal_array.NumericTypeCodeToGDALTypeCode(data.dtype)
+    dataset = mem_drv.Create('', cols, rows, bands, gdal_type)
+
+    # initialize geotransform
+    x_ps, y_ps = coords[1, 1] - coords[0, 0]
+    upper_corner_x = coords[0, 0, 0] - x_ps/2
```

Comment:
This assumes `centers`, right? IIRC, geotransform uses the corners of the pixels not the centers.

Reply:
OK got it, we are assuming here `coords` are center-coords, not edge coords. Should we make this clear somewhere? And, what if the user already has edge coordinates?